src/components/Search/SearchForm.jsx

```diff
@@ -23,6 +23,9 @@ class SearchForm extends React.Component {
     const { pathname } = window.location;
     if (params.q && pathname === '/search') {
       this.props.dispatchSearch(params.q);
+      this.setState({
```

Comment:
i think you can just remove the `     if (pathname === '/search') {` from the this.debouncedSetQuery() below

Reply:
Interestingly, when I tried that, I got an error saying debounce expects a function. 
![image](https://user-images.githubusercontent.com/12708206/36368464-5ced44a4-157d-11e8-8b75-2aaf06b25771.png)

I think the debounce that is done may be in an earlier context, where the function does not exist yet (from the props), and this check was added to only do it after the search context was received. 

https://stackoverflow.com/a/28046731 (Don't like to use stackoverflow as source, but this guy gives a decent explanantion)

I'm sorry if I've misunderstood something, I don't have a lot of experience with debouncing. Maybe this is not the issue at all.

Reply:
Try wrapping it in `if (this.props.dispatchSetQuery)` to ensure it's defined

Reply:
Oh, yeah, sure. Could do that. Is that fine from a design perspective?

Reply:
So that doesn't seem to be working either. It's not recognising the dispatch function as a function. 

I thought this issue was not related to debouncing, only involved setting state at component mount, since the props will have the query.

Reply:
yeah, looks like this will be fine.  Thanks!